docs/reference/content/driver/reference/fle.md

```diff
@@ -0,0 +1,166 @@
+
+{{< install artifactId="mongodb-mongocrypt" version="1.0.0-SNAPSHOT">}}
+
+If the jar fails to run there are separate jar files for specific architectures:
+
+#### RHEL 7.0*
+{{< install artifactId="mongodb-mongocrypt" version="1.0.0-SNAPSHOT" classifier="linux64-rhel70">}}
+
+#### OSX*
+{{< install artifactId="mongodb-mongocrypt" version="1.0.0-SNAPSHOT" classifier="osx">}}
+
+#### Windows*
+{{< install artifactId="mongodb-mongocrypt" version="1.0.0-SNAPSHOT" classifier="win64">}}
+
+#### Ubuntu 16.04
+{{< install artifactId="mongodb-mongocrypt" version="1.0.0-SNAPSHOT" classifier="linux64-ubuntu1604">}}
+
+
+* Distribution is included in the main `mongodb-mongocrypt` jar file.
+
+### mongocryptd configuration
+
+`libmongocrypt` requires the `mongocryptd` daemon / process to be running. A specific daemon / process uri can be configured in the 
+`AutoEncryptionSettings` class by setting `mongocryptdURI` in the `extraOptions`.
+
+{{% note class="important" %}}
+If not configured the driver will automatically try to start the daemon / process in: `/tmp/mongocryptd.sock`. This requires
+the `jnr.unixsocket` library to be installed.
+{{% /note %}}
+
+More information about libmongocrypt will soon be available from the official documentation.
+
+
+### Examples
+
+The following is a sample app that assumes key and schema have already been created in MongoDB and the `encryptedField` field is encrypted:
+
+```java
+import com.mongodb.AutoEncryptionSettings;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClients;
+import org.bson.Document;
+
+import java.security.SecureRandom;
+import java.util.Map;
+
+public class ClientSideEncryptionSimpleTest {
+
+    public static void main(String[] args) {
+
+        // This would have to be the same master key as was used to create the encryption key
+        var localMasterKey = new byte[96];
+        new SecureRandom().nextBytes(localMasterKey);
+
+        var kmsProviders = Map.of("local", Map.<String, Object>of("key", localMasterKey));
+        var keyVaultNamespace = "admin.datakeys";
+
+        var autoEncryptionSettings = AutoEncryptionSettings.builder()
+            .keyVaultNamespace(keyVaultNamespace)
+            .kmsProviders(kmsProviders)
+            .build();
+
+        var clientSettings = MongoClientSettings.builder()
+            .autoEncryptionSettings(autoEncryptionSettings)
+            .build();
+
+        var client = MongoClients.create(clientSettings);
+        var collection = client.getDatabase("test").getCollection("coll");
+        collection.drop(); // Clear old data
+
+        collection.insertOne(new Document("encryptedField", "123456789"));
+
+        System.out.println(collection.find().first().toJson());
+    }
+}
+```
+
+{{% note %}}
+Auto encryption is an **enterprise** only feature.
+{{% /note %}}
+
+The following example shows how to configure the `AutoEncryptionSettings` instance and set the json schema map to be used:
```

Comment:
It's also showing creation of a new encryption key

Reply:
Updated